src/protocol/ipa_prf/prf_eval/mod.rs

```diff
@@ -0,0 +1,191 @@
+use ipa_macros::Step;
+
+use crate::{
+    error::Error,
+    ff::{curve_points::RP25519, ec_prime_field::Fp25519},
+    protocol::{
+        basics::{Reveal, SecureMul},
+        context::Context,
+        prss::SharedRandomness,
+        RecordId,
+    },
+    secret_sharing::replicated::{semi_honest::AdditiveShare, ReplicatedSecretSharing},
+    seq_join::seq_try_join_all,
+};
+
+#[derive(Step)]
+pub(crate) enum Step {
+    PRFKeyGen,
+    GenRandomMask,
+    MultMaskWithPRFInput,
+    RevealR,
+    Revealz,
+}
+
+/// # Errors
+/// Propagates errors from multiplications
+pub async fn compute_match_key_pseudonym<C>(
+    sh_ctx: C,
+    prf_key: AdditiveShare<Fp25519>,
+    input_match_keys: Vec<AdditiveShare<Fp25519>>,
+) -> Result<Vec<u64>, Error>
+where
+    C: Context,
+{
+    let ctx = sh_ctx.set_total_records(input_match_keys.len());
+    let futures = input_match_keys
+        .iter()
+        .enumerate()
+        .map(|(i, x)| eval_dy_prf(ctx.clone(), i.into(), &prf_key, x));
+    seq_try_join_all(sh_ctx.active_work(), futures).await
+}
+
+impl From<AdditiveShare<Fp25519>> for AdditiveShare<RP25519> {
+    fn from(s: AdditiveShare<Fp25519>) -> Self {
+        AdditiveShare::new(RP25519::from(s.left()), RP25519::from(s.right()))
+    }
+}
+
+/// generates PRF key k as secret sharing over Fp25519
+pub fn gen_prf_key<C>(ctx: &C) -> AdditiveShare<Fp25519>
+where
+    C: Context,
+{
+    ctx.narrow(&Step::PRFKeyGen)
+        .prss()
+        .generate_replicated(u128::MAX - 100u128)
```

Comment:
this is very a suspicious code - I hope it is not used anywhere and can be deleted